exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsAggBatch.java

```diff
@@ -0,0 +1,256 @@
+    TypedFieldId[] keyOutputIdsArray = new TypedFieldId[keyOutputIds.size()];
+
+    keyExprs.toArray(keyExprsArray);
+    valueExprs.toArray(valueExprsArray);
+    keyOutputIds.toArray(keyOutputIdsArray);
+
+    setupIsSame(cg, keyExprsArray);
+    setupIsSameApart(cg, keyExprsArray);
+    addRecordValues(cg, valueExprsArray);
+    outputRecordKeys(cg, keyOutputIdsArray, keyExprsArray);
+    outputRecordKeysPrev(cg, keyOutputIdsArray, keyExprsArray);
+
+    cg.getBlock("resetValues")._return(JExpr.TRUE);
+    getIndex(cg);
+
+    container.buildSchema(SelectionVectorMode.NONE);
+    StreamingAggregator agg = context.getImplementationClass(cg);
+    agg.setup(oContext, incoming, this);
+    return agg;
+  }
+
+  private boolean isImplicitFileColumn(MaterializedField mf) {
+    return ImplicitColumnExplorer.initImplicitFileColumns(context.getOptions()).get(mf.getName()) != null;
+  }
+
+  protected StreamingAggregator createAggregatorInternal()
+      throws SchemaChangeException, ClassTransformationException, IOException {
+    container.clear();
+
+    List<LogicalExpression> keyExprs = Lists.newArrayList();
+    List<LogicalExpression> valueExprs = Lists.newArrayList();
+    List<TypedFieldId> keyOutputIds = Lists.newArrayList();
+    GregorianCalendar calendar = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+    calendar.setTimeInMillis(System.currentTimeMillis());
+
+    if (this.getPopConfig() instanceof StatisticsAggregate
+        && (((StatisticsAggregate) this.getPopConfig()).getPhase() == OperatorPhase.PHASE_1of1
+            || ((StatisticsAggregate) this.getPopConfig()).getPhase() == OperatorPhase.PHASE_2of2)) {
+      createKeyColumn("schema",
+          ValueExpressions.getBigInt(schema++),
+          keyExprs,
+          keyOutputIds
+      );
+      createKeyColumn("computed",
+          ValueExpressions.getDate(calendar),
+          keyExprs,
+          keyOutputIds
+      );
+    }
+
+    MapVector cparent = new MapVector("column", oContext.getAllocator(), null);
+    container.add(cparent);
+    for (MaterializedField mf : incoming.getSchema()) {
+      // Ignore implicit columns
+      if (!isImplicitFileColumn(mf)) {
+        createNestedKeyColumn(
+            cparent,
+            mf.getLastName(),
+            ValueExpressions.getChar(mf.getLastName()),
+            keyExprs,
+            keyOutputIds
+        );
+      }
+    }
+
+    for (String func : functions) {
+      MapVector parent = new MapVector(func, oContext.getAllocator(), null);
+      container.add(parent);
+
+      for (MaterializedField mf : incoming.getSchema()) {
+        if (!isImplicitFileColumn(mf)) {
+          List<LogicalExpression> args = Lists.newArrayList();
+          args.add(SchemaPath.getSimplePath(mf.getPath()));
+          LogicalExpression call = FunctionCallFactory.createExpression(func, args);
+          addMapVector(mf.getLastName(), parent, call, valueExprs);
```

Comment:
The current design does not handle nested maps. This limitation would be documented. We throw an error now saying that maps as input are not supported.